### PR TITLE
7-POST api/articles/:article_id/comments 

### PR DESCRIPTION
**Pull request by melissa-linsc**
Work on 7-POST api/articles/:article_id/comments and some feedback changes from previous task

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -170,6 +167,77 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        .expect(201)
+        .send(newComment)
+        .then(({body}) => {
+                expect(body.comment).toMatchObject({
+                    comment_id: 19,
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    article_id: 1,
+                    author: expect.any(String),
+                    body: expect.any(String)
```
Comment (nc-scarlett-adams) [inline __tests__/app.test.js:186]: For this newComment, as we know what we post, so author and body are better to be more specific

File: __tests__/app.test.js [hunk 2]
```diff
@@ -170,6 +167,77 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+            body: "This morning, I showered for nine minutes.", 
+            username: "butter_bridge",
+            extraKey: "extra key"}
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send(newComment)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid Request Body')
+        })
```
Comment (nc-scarlett-adams) [inline __tests__/app.test.js:225]: It's good that you've handled the extra key, but it can simply be ignored if there're extra properties

File: __tests__/app.test.js [hunk 3]
```diff
@@ -170,6 +167,77 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+    });
+    test('should return a 400 Invalid Request Body when the request body has more keys than required', () => {
```
Comment (nc-scarlett-adams) [inline __tests__/app.test.js:227]: This test and the one above seem have the descriptions in wrong order

File: __tests__/app.test.js [hunk 4]
```diff
@@ -170,6 +167,77 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send(newComment)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid Request Body')
+        })
+    });
+});
+
+
```
Comment (nc-scarlett-adams) [inline __tests__/app.test.js:240]: A test 404 - username not found needs to be added here

File: controllers/articles.controller.js [hunk 5]
```diff
@@ -26,4 +27,25 @@ const getCommentsByArticleId = (req,res,next) => {
...
-module.exports = {getArticleById, getArticles, getCommentsByArticleId}
\ No newline at end of file
+const postCommentsByArticleId = (req,res,next) => {
+
+    const articleId = req.params.article_id
+    const newComment = req.body
+
+    if (!newComment.body || !newComment.username || Object.keys(newComment).length !==2) {
+        return next({status:400, msg: 'Invalid Request Body'})
+    }
```
Comment (nc-scarlett-adams) [inline controllers/articles.controller.js:37]: Probably better to have this part in your model and actually throw an error back (using Promise.reject) instead of invoke next with an object

**Review by nc-scarlett-adams — COMMENTED** [review]
Hi Melissa, a couple of changes need to make. Adding one more test for none existent username
